aws/resource_aws_rds_cluster_endpoint.go

```diff
@@ -0,0 +1,267 @@
+		input.StaticMembers = make([]*string, 0)
+	}
+
+	_, err := conn.ModifyDBClusterEndpoint(input)
+	if err != nil {
+		return fmt.Errorf("Error modifying RDS Cluster Endpoint: %s", err)
+	}
+
+	return resourceAwsRDSClusterEndpointRead(d, meta)
+}
+
+func resourceAwsRDSClusterEndpointDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+	input := &rds.DeleteDBClusterEndpointInput{
+		DBClusterEndpointIdentifier: aws.String(d.Id()),
+	}
+	_, err := conn.DeleteDBClusterEndpoint(input)
+	if err != nil {
+		return fmt.Errorf("Error deleting RDS Cluster Endpoint: %s", err)
+	}
+
+	if err := resourceAwsRDSClusterEndpointWaitForDestroy(d.Timeout(schema.TimeoutDelete), d.Id(), conn); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsRDSClusterEndpointWaitForDestroy(timeout time.Duration, id string, conn *rds.RDS) error {
+	log.Printf("Waiting for RDS Cluster Endpoint %s to be deleted...", id)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"available", "deleting"},
+		Target:     []string{"destroyed"},
+		Refresh:    DBClusterEndpointStateRefreshFunc(conn, id),
+		Timeout:    timeout,
+		Delay:      AWSRDSClusterEndpointRetryDelay,
+		MinTimeout: AWSRDSClusterEndpointRetryMinTimeout,
+	}
+	_, err := stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for RDS Cluster Endpoint (%s) to be deleted: %v", id, err)
+	}
+	return nil
+}
+
+func resourceAwsRDSClusterEndpointWaitForAvailable(timeout time.Duration, id string, conn *rds.RDS) error {
+	log.Printf("Waiting for RDS Cluster Endpoint %s to become available...", id)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"creating"},
+		Target:     []string{"available"},
+		Refresh:    DBClusterEndpointStateRefreshFunc(conn, id),
+		Timeout:    timeout,
+		Delay:      AWSRDSClusterEndpointRetryDelay,
+		MinTimeout: AWSRDSClusterEndpointRetryMinTimeout,
+	}
+
+	_, err := stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for RDS Cluster Endpoint (%s) to be ready: %v", id, err)
+	}
+	return nil
+}
+
+func DBClusterEndpointStateRefreshFunc(conn *rds.RDS, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		emptyResp := &rds.DescribeDBClusterEndpointsOutput{}
+
+		resp, err := conn.DescribeDBClusterEndpoints(
+			&rds.DescribeDBClusterEndpointsInput{
+				DBClusterEndpointIdentifier: aws.String(id),
+			})
+		if err != nil {
+			if isAWSErr(err, rds.ErrCodeDBClusterNotFoundFault, "") {
+				return emptyResp, "destroyed", nil
+			} else if resp != nil && len(resp.DBClusterEndpoints) == 0 {
+				return emptyResp, "destroyed", nil
+			} else {
+				return emptyResp, "", fmt.Errorf("Error on refresh: %+v", err)
+			}
+		}
+
+		if resp == nil || resp.DBClusterEndpoints == nil || len(resp.DBClusterEndpoints) == 0 {
+			return emptyResp, "destroyed", nil
+		}
+
+		return resp.DBClusterEndpoints[0], *resp.DBClusterEndpoints[0].Status, nil
```

Comment:
Nit: To prevent potential panics we should use the SDK provided helper function for dereferencing the `Status`, e.g.

```suggestion
		return resp.DBClusterEndpoints[0], aws.StringValue(resp.DBClusterEndpoints[0].Status), nil
```